### ISSUE TITLE
Roborock S5 Max "Unknown error code"

### ISSUE THREAD

**Issue by Shariner** [arner/roborock-glados#2]
Hey,

this voice pack seems to be incompatible with the S5 Max. If I want to upload the .pkg file, I will get a "RoborockVoicePackManagementCapability: Failed to install Voicepack. Unknown error code 3" error. Sometimes it's Unknown error code 6. - No idea what this error code could be.

I am a bit confused. The original sound files have an .ogg extension, not .wav. 

I also thought the voice pack could be too big to upload, I cleaned up logs to free up some space, no luck. Storage shouldn't be a problem.

Then i wanted to replace the voice files on the file system, they are located at /mnt/resources/audio_custom/sounds - but the file system is read-only, even with the root user. 

Last idea would be to generate an own sound package with .ogg files instead of .wav and pack them. Maybe beginning with one or 2 replaced sounds to avoid an issue with file size, but currently I have no time to get a linux distribution running to execute the files and clone the repo.

Maybe I will update this, if I am able to fix the issue.

Otherwise: Let me know if your roborock can transform into Glados to serve cake.

**Comment by Shariner** [comment]
So after setting up another Linux Dist, installing tools etc. no luck.

I guess the S5 Max needs a special password for the .pkg file or a correct md5 hash. 

There is a "info" file in the sound resources, providing md5 checkusms for any official sound packages. I can't edit this file, but I think the robot will hash the uploaded package and compare the md5 against the list.

I also did not found a single custom voice package or instruction for the S5 Max.

**Comment by Thompson9433** [comment]
I have the same issue with Roborock S8, did you find solution? I have .ogg file and i packed them into .pkg file, but this isnt working either.

**Comment by Shariner** [comment]
> I have the same issue with Roborock S8, did you find solution? I have .ogg file and i packed them into .pkg file, but this isnt working either.

Yeah there is a valetudo Telegramm group with very much information inside. The S5 Max is build differently and uses hashed voice packages in the squashfs system, therefore it’s not possible to simply install an custom voice package next to the official ones. 

It could be, that the same applies to the S8. 

So we ran into to issues:
1. create and flash the hashed voice package 
2. Storage size as the read only partition is very small and the glados voice package is very large

There are ways to get custom voices into the S5 Max, example this project https://github.com/urholaukkarinen/voicegen
But it was too deep inside the system for me, stopped trying before I will break the S5 Max.
